### PR TITLE
feat: split the whole map into many chunks (16*16)

### PR DESCRIPTION
**Pull request by moe-moe-pupil**
### Rationale for implementing this:
split the whole map into that many chunks(16 * 16)

### Referenced Issue:
editing a big map is super laggy

### Demo game JSON:
[expandaria.txt](https://github.com/user-attachments/files/16687250/expandaria.txt)

### CODE REVIEW

File: ts/src/renderer/three/VoxelEditor.ts [hunk 1]
```diff
@@ -489,20 +491,24 @@ class VoxelEditor {
...
 		this.currentLayerIndex = value;
-		voxels.meshes[value].visible = true;
+
+		Object.values(voxels.meshes[value]).forEach((mesh) => (mesh.visible = true));
 	}
 
 	hideLayer(layer: number, state: boolean): void {
-		Renderer.Three.getVoxels().meshes[layer].visible = !state;
+		const voxels = Renderer.Three.getVoxels();
+		Object.values(voxels.meshes[layer]).forEach((mesh) => (mesh.visible = !state));
```
Comment (nickyvanurk) [inline ts/src/renderer/three/VoxelEditor.ts:500]: Is meshes a map? Please just use .values() or .entries() and a normal for...of loop instead of .forEach.

File: ts/src/renderer/three/VoxelEditor.ts [hunk 2]
```diff
@@ -489,20 +491,24 @@ class VoxelEditor {
...
 	}
 
 	showAllLayers(): void {
 		const voxels = Renderer.Three.getVoxels();
-		for (let i = 0; i < voxels.meshes.length; i++) {
-			if (voxels?.meshes[i]?.visible === false) {
-				this.hideLayer(i, false);
-			}
-		}
+		Object.entries(voxels.meshes).forEach(([layer, chunkKey]) =>
```
Comment (nickyvanurk) [inline ts/src/renderer/three/VoxelEditor.ts:505]: Same

**Review by nickyvanurk — COMMENTED** [review]
Hard to follow the code because I haven't worked on this for a while. Didn't see any major things that stand out to me. Could use a bit more empty lines between pieces of code though (like between if statements, sections of code that belong together etc). It's hard to read when all the code is clumped up without empty lines.

File: ts/src/renderer/three/VoxelEditor.ts (hunk 1 above)
Comment (moe-moe-pupil) [reply ts/src/renderer/three/VoxelEditor.ts]: done xD
